pkg/admissioncontroller/webhook/admission/authenticationconfig/handler.go

```diff
@@ -0,0 +1,235 @@
+	}
+
+	if errCode, err := validateAuthenticaionConfigurationSemantics(authenticationConfiguration); err != nil {
+		return admission.Errored(errCode, err)
+	}
+
+	return admissionwebhook.Allowed("referenced authentication configuration is valid")
+}
+
+func (h *Handler) admitConfigMap(ctx context.Context, request admission.Request) admission.Response {
+	var (
+		oldCm = &corev1.ConfigMap{}
+		cm    = &corev1.ConfigMap{}
+	)
+
+	if request.Operation != admissionv1.Update {
+		return admissionwebhook.Allowed("operation is not update")
+	}
+
+	if err := h.Decoder.Decode(request, cm); err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	// lookup if configmap is referenced by any shoot in the same namespace
+	shootList := &gardencorev1beta1.ShootList{}
+	if err := h.Client.List(ctx, shootList, client.InNamespace(request.Namespace)); err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	var configMapIsReferenced bool
+	for _, shoot := range shootList.Items {
+		if v1beta1helper.GetShootAuthenticationConfigurationConfigMapName(shoot.Spec.Kubernetes.KubeAPIServer) == request.Name {
+			configMapIsReferenced = true
+			break
+		}
+	}
+
+	if !configMapIsReferenced {
+		return admissionwebhook.Allowed("configmap is not referenced by a Shoot")
+	}
+
+	authenticaionConfiguration, err := getAuthenticationConfiguration(cm)
+	if err != nil {
+		return admission.Errored(http.StatusUnprocessableEntity, err)
+	}
+
+	if err = h.getOldObject(request, oldCm); err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+	oldAuthenticationConfiguration, ok := oldCm.Data[authenticationConfigurationCmDataKey]
+	if ok && oldAuthenticationConfiguration == authenticaionConfiguration {
+		return admissionwebhook.Allowed("authentication configuration not changed")
+	}
+
+	errCode, err := validateAuthenticaionConfigurationSemantics(authenticaionConfiguration)
+	if err != nil {
+		return admission.Errored(errCode, err)
+	}
+
+	return admissionwebhook.Allowed("configmap change is valid")
+}
+
+func (h *Handler) getOldObject(request admission.Request, oldObj runtime.Object) error {
+	if len(request.OldObject.Raw) != 0 {
+		return h.Decoder.DecodeRaw(request.OldObject, oldObj)
+	}
+	return errors.New("could not find old object")
+}
+
+func validateAuthenticaionConfigurationSemantics(authenticaionConfiguration string) (errCode int32, err error) {
+	authenticationConfigurationObj, schemaVersion, err := configDecoder.Decode([]byte(authenticaionConfiguration), nil, nil)
+	if err != nil {
+		return http.StatusUnprocessableEntity, fmt.Errorf("failed to decode the provided authentication configuration: %w", err)
+	}
+	authenticationCOnfigurationInternal, ok := authenticationConfigurationObj.(*apiserver_internal.AuthenticationConfiguration)
```

Comment:
```suggestion
	authenticationConfigurationInternal, ok := authenticationConfigurationObj.(*apiserver_internal.AuthenticationConfiguration)
```

Not a strong opinion on this, but I would prefer a shorter names for these variables. Something like `authConfig` and `authConfigObj`